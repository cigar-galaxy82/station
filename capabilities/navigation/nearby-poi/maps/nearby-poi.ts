/* eslint-disable jest/no-export */

import { RecordingProcessFunctions, SuperfaceTest } from '@superfaceai/testing-lib';

export const nearbyPoiTest = (
  provider: string,
  options?: {
    supertestHooks?: RecordingProcessFunctions
  }
): void => {
  describe(`navigation/nearby-poi/${provider}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest();
    });

    describe('find nearby pois', () => {
      it('it should find astronomy cafe', async () => {
        const input = {
          center: {
            latitude: 51.477,
            longitude: 0.0
          },
          radius: 100,
          categories: ['CAFE']
        };

        await expect(
          superface.run({
            profile: 'navigation/nearby-poi',
            provider,
            useCase: 'NearbyPoi',
            input,
          }, options?.supertestHooks)
        ).resolves.toMatchSnapshot();
      });
    });
  });
};