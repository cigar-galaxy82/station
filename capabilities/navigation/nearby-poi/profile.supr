"""
Find nearby points of interest.

Find points of interest near the given location using map services.
"""
name = "navigation/nearby-poi"
version = "1.0"

"""
Find nearby points of interest

Find points of interest near the given location.
"""
usecase NearbyPoi {
	input {
		"Center of the search"
		center! Coordinates!
		"Radius of the search"
		radius! number!
		"""
		Optional categories filter

		Points belonging to at least one of these categories are returned
		"""
		categories [InterestCategory!]
	}

	result [
		{
			"Coordinates of this point"
			coordinates! Coordinates!
			"Name of the point of interest"
			name! string!
			"Categories this point belongs to"
			categories! [InterestCategory!]
		}
	]

	error {
		"Human-readable status description"
		status! string!
		"Human-readable error message"
		message! string!
	}
}

model Coordinates {
	latitude! number!
	longitude! number!
}

model InterestCategory enum {
	RESTAURANT,
	CAFE,
	BAR
	// TODO: More categories
}