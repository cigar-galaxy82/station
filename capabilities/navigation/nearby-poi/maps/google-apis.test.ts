import { nearbyPoiTest } from './nearby-poi';

nearbyPoiTest(
  'google-apis',
  {
    supertestHooks: {
      afterRecordingLoad: (scopes) => {
        scopes.forEach(scope => {
          scope.filteringPath(
            /key=[^&]*/g,
            'key=credentials-removed-to-keep-them-secure'
          );
        });
      }
    }
  }
);
