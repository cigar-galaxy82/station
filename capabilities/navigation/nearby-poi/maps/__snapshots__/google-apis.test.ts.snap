// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation/nearby-poi/google-apis find nearby pois it should find astronomy cafe 1`] = `
Object {
  "value": Array [
    Object {
      "categories": Array [
        "CAFE",
      ],
      "coordinates": Object {
        "latitude": 51.47674399999999,
        "longitude": -0.0005606,
      },
      "name": "Astronomy Cafe",
    },
    Object {
      "categories": Array [
        "CAFE",
      ],
      "coordinates": Object {
        "latitude": 51.4766838,
        "longitude": -0.0007296,
      },
      "name": "Gagarin Terrace",
    },
    Object {
      "categories": Array [
        "CAFE",
      ],
      "coordinates": Object {
        "latitude": 51.4767028,
        "longitude": 0.0006828000000000001,
      },
      "name": "The Pavilion Cafe",
    },
  ],
}
`;
